src/lib/http/http_server.pl

```diff
@@ -1,51 +1,51 @@
 /* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
    Written in December 2020 by Adrián Arroyo (adrian.arroyocalle@gmail.com)
    Updated in March 2022 by Adrián Arroyo to use the Hyper backend
-   Part of Scryer Prolog
-
-   This library provides an starting point to build HTTP server based applications.
-   It is based on Hyper, which allows for HTTP/1.0, HTTP/1.1 and HTTP/2. However,
-   some advanced features that Hyper provides are still not accesible.
-
-   Usage
-   ==========
-   The main predicate of the library is http_listen/2, which needs a port number
-    (usually 80) and a list of handlers. A handler is a compound term with the functor
-   as one HTTP method (in lowercase) and followed by a Route Match and a predicate
-   which will handle the call.
-
-   text_handler(Request, Response) :-
-    http_status_code(Response, 200),
-    http_body(Response, text("Welcome to Scryer Prolog!")).
-
-   parameter_handler(User, Request, Response) :-
-    http_body(Response, text(User)).
-
-   http_listen(7890, [
-        get(echo, text_handler),           % GET /echo
-        post(user/User, parameter_handler(User)) % POST /user/<User>
-   ]).
-
-   Every handler predicate will have at least 2-arity, with Request and Response.
-   Although you can work directly with http_request and http_response terms, it is
-   recommeded to use the helper predicates, which are easier to understand and cleaner:
-   - http_headers(Response/Request, Headers)
-   - http_status_code(Responde, StatusCode)
-   - http_body(Response/Request, text(Body))
-   - http_body(Response/Request, binary(Body))
-   - http_body(Request, form(Form))
-   - http_body(Response, file(Filename))
-   - http_redirect(Response, Url)
-   - http_query(Request, QueryName, QueryValue)
+   Part of Scryer Prolog.
+   I place this code in the public domain. Use it in any way you want.
+*/
+
+/** This library provides an starting point to build HTTP server based applications.
+It is based on Hyper, which allows for HTTP/1.0, HTTP/1.1 and HTTP/2. However,
+some advanced features that Hyper provides are still not accesible.
+
+## Usage
+
+The main predicate of the library is http\_listen/2, which needs a port number
```

Comment:
Why must `_` sometimes be escaped as `\_`, and sometimes not?

Reply:
Top predicate header is not Markdown and code blocks inside Markdown do not have have emphasis. But outside, lots of Prolog predicates trigger emphasis... But that depends on the implementation you're using. My implementation, as well as original Markdown.pl (try here: https://daringfireball.net/projects/markdown/dingus) will treat set_prolog_flag/2 as set-OPEN EMPHASIS-prolog-CLOSE EMPHASIS-flag/2. However, as you see GitHub Flavoured Markdown does not... I might try to implement CommonMark emphasis rules which seem more popular. Although I'm starting to think using Markdown is not a good idea